case-lib/lib.sh

```diff
@@ -693,6 +693,19 @@ ipc4_used()
     return 0
 }
 
+is_ipc4_zephyr(){
+    local firmware_path firmware_name znum
+    fw_path=/sys/module/snd_sof_pci/parameters/fw_path
```

Comment:
The above doesn't work if fw_path is not set. At least I had a local setup where I just use the standard path assumed by kernel for IPC4 FW. 

Maybe if fw_path is NULL, check for /lib/firmware/intel/avs/"$platf"/dsp_basefw.bin . This is what is in SOF kernel driver by default unless fw_path is passed as a module parameter. 

Above can be done in a seperate follow-up PR.

Reply:
Yes, I can do that, but we need to solve one problem, the full `fw_path` should be `/lib/firmware/intel/avs/"$platf"/"$key_type"/dsp_basefw.bin`. 

For example, the full fw path of TGLU_UP_HDA_IPC4ZPH is `/lib/firmware/intel/avs/tgl/community/dsp_basefw.bin`.  The key_type is defined in community boards :https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/sof-pci-dev.c#:~:text=/*%20all%20Up,%7D%3B.

I can use the DMI info to get the key_type, but better to let the kernel driver to expose the fw_path as Marc suggested below.